docs/proposals/optional_application_loging.md

```diff
@@ -0,0 +1,106 @@
+Optional Application Logging
+============================
+
+The EFK aggregated logging stack by default collects application logs with
+fluentd from STDOUT of each container managed by OpenShift. Currently only way
+of opting out is based on a node label and ansible variable
+`openshift_logging_fluentd_nodeselector` as described in the
+[documentation](https://docs.openshift.org/latest/install_config/aggregate_logging.html).
+This proposal discusses some ways of a finer grained level of control over
+which container logs are stored in Elasticsearch.
+
+
+## Fluentd pipeline
+
+The container logs can be obtained from one of these supported logging drivers
+- json-file - logs read from `/var/log/containers/*` mounted in fluentd
+              container
+- journald
+
+Latest development shifted recommended docker driver from journald to json-file,
+therefore, the proposed implementation has focused on usage of json-file as
+well. Rationale being, journald currently doesn't match performance expectations
+and desired throttling control for larger clusters.
+
+Each container on the host has its own json structured log file
+- `/var/log/containers/[pod]_[namespace]_[container]-[docker_id].log`
+
+Every line is treated as a separate log message and processed in the cascade
+of fluentd filters with currently only supported sink Elasticsearch.
+A log line doesn't contain OpenShift metadata, this is added to the message
+in the fluentd `k8_metadata_filter` plugin.
+
+#### Dynamic Control of Optional Logging
+
+By 'dynamic control' we understand the ability of an operator or a user to
+opt-in or opt-out certain OpenShift object from aggregated logging without
+having to redeploy fluentd.
+
+This approach has immediate effect, offers very granular level of control,
+is flexible, and doesn't bring any measurable latency for other containers logs.
+It requires additional resources and brings an overhead to the API server, where
+the fluentd input plugin has to `watch` certain API endpoints.
+
+Following schema describes how this can be implemented as an opt-out kind:
+- a user will label a namespace or a pod with `aggregated_logging_opt_out=true`
```

Comment:
Write in the present:
a user labels a namespace or a pod with `aggregated_logging_opt_out=true`